src/extension.ts

```diff
@@ -1137,81 +1137,86 @@ export async function activate(context: vscode.ExtensionContext) {
 						);
 					}
 
-					const rawPath =
-						(uri || vscode.window.activeTextEditor?.document.uri)
-							?.path ?? null;
+					const uri =
+						(uriArg ||
+							vscode.window.activeTextEditor?.document.uri) ??
+						null;
 
-					if (rawPath === null) {
+					if (uri === null || !uri.path) {
 						return;
 					}
 
-					const mostRecentCodemodHash =
-						codemodListWebviewProvider.getMostRecentCodemodHash();
+					const codemodList = await engineService.getCodemodList();
 
-					if (mostRecentCodemodHash === null) {
-						return;
-					}
+					// order: least recent to most recent
+					const top3RecentCodemodHashes = codemodListWebviewProvider
+						.getRecentCodemodHashes()
+						.slice(-3);
 
-					await codemodListWebviewProvider.updateExecutionPath({
-						newPath: rawPath,
-						codemodHash: mostRecentCodemodHash,
-						fromVSCodeCommand: true,
-					});
-
-					vscode.commands.executeCommand(
-						'workbench.view.extension.intuitaViewId',
+					const top3RecentCodemods = codemodList.filter((codemod) =>
+						top3RecentCodemodHashes.includes(codemod.hashDigest),
 					);
-					setTimeout(() => {
-						messageBus.publish({
-							kind: MessageKind.focusCodemod,
-							codemodHashDigest: mostRecentCodemodHash,
-						});
-					}, 500);
-				} catch (error) {
-					vscode.window.showErrorMessage(
-						error instanceof Error ? error.message : String(error),
-					);
-				}
-			},
-		),
-	);
-
-	context.subscriptions.push(
-		vscode.commands.registerCommand(
-			'intuita.executeCodemodWithinPath',
-			async (uri: vscode.Uri) => {
-				try {
-					const { storageUri } = context;
-
-					if (!storageUri) {
-						throw new Error(
-							'No storage URI, aborting the command.',
+					top3RecentCodemods.sort((a, b) => {
+						return (
+							top3RecentCodemodHashes.indexOf(b.hashDigest) -
+							top3RecentCodemodHashes.indexOf(a.hashDigest)
 						);
-					}
-
-					const codemodList = await engineService.getCodemodList();
-
+					});
+					const sortedCodemodList = [
+						...top3RecentCodemods,
+						...codemodList.filter(
+							(codemod) =>
+								!top3RecentCodemods
```

Comment:
you can actually check if `top3RecentCodemodHashes` contains `codemod.hash` for simplicity.